measurements/api/measurements.py

```diff
@@ -668,10 +643,16 @@ def coal(colname):
 
         raise exc
 
-    # For each report_id / input tuple, we want at most one entry from the
-    # traditional pipeline and one from fastpath, merged together
+    # For each report_id / input tuple, we want at most one entry. Measurements
+    # from mr_table and fastpath has already been merged by the FULL OUTER JOIN
+    # but we have duplicate msmts sharing the same report_id / input.
     results = _merge_results(tmpresults)
 
+    # Replace the special value INULL for "input" with None
+    for i, r in enumerate(results):
+        if r["input"] == INULL:
+            results[i]["input"] = None
```

Comment:
Is this done because otherwise the explorer links will not work as the input is set to ""?

Maybe it's useful to also mention that INULL is equal to the empty string in the docstring.

Reply:
https://github.com/ooni/api/pull/171/files/7489bda785d4d3a193a8252262b5be0fb0f75a1e#diff-a6bf0b1b75ef5e2f06d17e0a4e9bbc4aR367  it's described here. Currently the empty string seems to work safely but we could use a different magic value if needed.